pkg/controller/installation/core_controller.go

```diff
@@ -241,12 +241,26 @@ type ReconcileInstallation struct {
 	migrationChecked     bool
 }
 
-// GetInstallation returns the current installation resource.
-func GetInstallation(ctx context.Context, client client.Client) (*operator.Installation, error) {
+// GetInstallation returns an installation spec for use by other controllers.
+func GetInstallation(ctx context.Context, client client.Client) (*operator.InstallationSpec, error) {
 	// Fetch the Installation instance. We only support a single instance named "default".
-	instance := &operator.Installation{}
-	err := client.Get(ctx, utils.DefaultInstanceKey, instance)
-	return instance, err
+	instance := operator.Installation{}
+	if err := client.Get(ctx, utils.DefaultInstanceKey, &instance); err != nil {
+		return nil, err
+	}
+
+	// Return the computed spec in status as it both includes the 'overlay' installation details, but
+	// also represents the actual live installation details. This is important as other Enterprise controllers shouldn't
+	// launch pods until the core controller has completed it's transition to enterprise, which is indicated via the variant field
+	// in the status resource.
+	// Note that we don't need to merge data from the spec field because it basically already is for our purposes, since
+	// spec.computed isn't written until Installation has made it to that point.
+	// However we do run a nil check on Computed only because historically, this function never should return nil
+	// TODO: change to return reference instead of pointer
+	if instance.Status.Computed != nil {
+		return instance.Status.Computed, nil
+	}
+	return &operator.InstallationSpec{}, nil
```

Comment:
@tmjd this bit of code still has me unsure. The `Spec` field was always a literal. But because `Status.Computed` is a pointer, we have introduced a new code path where there was no error getting the Installation but it's status is nil.

You'll notice in the original code, there's no way to receive `nil, nil`. So I've replicated that here by returning an empty spec even when there's no status.

Reply:
What would you think of returning an IsNotFound error if Status.Computed is not set?